codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -0,0 +1,642 @@
+            "E" to errorSymbol,
+            "serialize_error" to serverSerializeError(operationShape)
+        )
+    }
+
+    /*
+     * TODO: implement straming traits
+     */
+    private fun RustWriter.renderStreamingTraits(
+        operationName: String,
+        outputSymbol: Symbol,
+        operationShape: OperationShape
+    ) {
+        logger.warning(
+            "[rust-server-codegen] $operationShape: streaming trait is not yet implemented"
+        )
+    }
+
+    private fun serverParseRequest(operationShape: OperationShape): RuntimeType {
+        val fnName = "parse_${operationShape.id.name.toSnakeCase()}_request"
+        val inputShape = operationShape.inputShape(model)
+        val inputSymbol = symbolProvider.toSymbol(inputShape)
+        return RuntimeType.forInlineFun(fnName, operationDeserModule) {
+            Attribute.Custom("allow(clippy::unnecessary_wraps)").render(it)
+            it.rustBlockTemplate(
+                "pub fn $fnName(request: &#{Http}::Request<#{Bytes}>) -> std::result::Result<#{I}, #{Error}>",
+                *codegenScope,
+                "I" to inputSymbol,
+            ) {
+                withBlock("Ok({", "})") {
+                    serverRenderShapeParser(
+                        operationShape,
+                        inputShape,
+                        httpBindingResolver.requestBindings(operationShape),
+                    )
+                }
+            }
+        }
+    }
+
+    private fun serverSerializeResponse(operationShape: OperationShape): RuntimeType {
+        val fnName = "serialize_${operationShape.id.name.toSnakeCase()}_response"
+        val outputShape = operationShape.outputShape(model)
+        val outputSymbol = symbolProvider.toSymbol(outputShape)
+        return RuntimeType.forInlineFun(fnName, operationSerModule) {
+            Attribute.Custom("allow(clippy::unnecessary_wraps)").render(it)
+            it.rustBlockTemplate(
+                "pub fn $fnName(output: &#{O}) -> std::result::Result<#{Http}::Response<#{Bytes}>, #{Error}>",
+                *codegenScope,
+                "O" to outputSymbol,
+            ) {
+                withBlock("Ok({", "})") {
+                    serverRenderShapeResponseSerializer(
+                        operationShape,
+                        httpBindingResolver.responseBindings(operationShape),
+                    )
+                }
+            }
+        }
+    }
+
+    private fun serverSerializeError(operationShape: OperationShape): RuntimeType {
+        val fnName = "serialize_${operationShape.id.name.toSnakeCase()}_error"
+        val errorSymbol = operationShape.errorSymbol(symbolProvider)
+        return RuntimeType.forInlineFun(fnName, operationSerModule) {
+            Attribute.Custom("allow(clippy::unnecessary_wraps)").render(it)
+            it.rustBlockTemplate(
+                "pub fn $fnName(error: &#{E}Kind) -> std::result::Result<#{Http}::Response<#{Bytes}>, #{Error}>",
```

Comment:
we should probably add a helper to the error generator that can make the error kind symbol...